From f21d61722cfe3b299b422d632539c103beda1c76 Mon Sep 17 00:00:00 2001
From: "Nikunj A. Dadhania" <nikunj.dadhania@intel.com>
Date: Mon, 4 Feb 2019 01:53:09 -0800
Subject: [PATCH 1248/1374] usb: dwc3: gadget: use req->needs_extra_trb flag

backport commit 25ad17d692ad54c3c33b2a31e5ce2a82e38de14e had
replaced 'needs_extra_trb' with 'unaligned' and 'zero' members

Restore it back to 'needs_extra_trb' and fix compile failure

Change-Id: Ic5559f0908a4faa88b2e88371d5d8dc7f4c8dbb9
Signed-off-by: Nikunj A. Dadhania <nikunj.dadhania@intel.com>
---
 drivers/usb/dwc3/gadget.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/usb/dwc3/gadget.c b/drivers/usb/dwc3/gadget.c
index c3294a2..b28409e 100644
--- a/drivers/usb/dwc3/gadget.c
+++ b/drivers/usb/dwc3/gadget.c
@@ -178,8 +178,7 @@ static void dwc3_gadget_del_and_unmap_request(struct dwc3_ep *dep,
 	req->started = false;
 	list_del(&req->list);
 	req->remaining = 0;
-	req->unaligned = false;
-	req->zero = false;
+	req->needs_extra_trb = false;
 
 	if (req->request.status == -EINPROGRESS)
 		req->request.status = status;
-- 
2.7.4

